packages/react-components/eslint-plugin-react-components/src/index.ts

```diff
@@ -0,0 +1,16 @@
+import * as fs from 'node:fs';
+import * as path from 'node:path';
+
+const pkg = JSON.parse(fs.readFileSync(path.join(__dirname, '../package.json'), 'utf8'));
+
+const plugin = {
+  meta: {
+    name: pkg.name,
+    version: pkg.version,
+  },
+  rules: {
+    // add rules here
+  },
+};
+
+export default plugin;
```

Comment:
not sure if this will work. we need this to be transpiled to `module.exports` can you please double check ? ty

Reply:
our workspace eslint-rules explicitly use `module.exports` - we should follow that I guess ( also other oss plugins do that )

Reply:
I'm not sure, but official docs doesn't say ESM is not supported 

<img width="1433" alt="image" src="https://github.com/user-attachments/assets/f795945c-fc84-49df-a484-6254191638e1">

Reply:
Switching to `module.exports` breaks `generate-api` output, and it returns only `export default {}`. And the same story with TS declarations.

Reply:
I would need to try out all the things myself I'm afraid :) It's always such a minefield!

I would _think_ though, worst case scenario, it ends up as `exports.default` in the transpiled JS, but because it's "just JS" in ESLint flat config, and you control the resolution, you would just need to do something like:

```js
const { default: myPlugin } = require('path/to/my-plugin');

```

Naturally you could take it way further and try and do dual ESM and CJS builds etc, but yeah to me the above seems fine...